packages/core/src/shared/utilities/workspaceUtils.ts

```diff
@@ -294,6 +295,35 @@ async function filterOutGitignoredFiles(rootPath: string, files: vscode.Uri[]):
     return gitIgnoreFilter.filterFiles(files)
 }
 
+/**
+ * Get the size of the currently sleected workspace and return it in bytes
+ * @param sourcePaths the paths where collection starts
+ * @param workspaceFolders the current workspace folders opened
+ * @param respectGitIgnore whether to respect gitignore file
+ * @returns the size of the workspace in bytes
+ *
+ */
+export async function getWorkspaceSize(
+    sourcePaths: string[],
+    workspaceFolders: CurrentWsFolders,
+    respectGitIgnore: boolean = true
+): Promise<number> {
+    let totalSizeBytes = 0
+    for (const rootPath of sourcePaths) {
+        const allFiles = await vscode.workspace.findFiles(
+            new vscode.RelativePattern(rootPath, '**'),
+            getExcludePattern()
+        )
+        const files = respectGitIgnore ? await filterOutGitignoredFiles(rootPath, allFiles) : allFiles
+
+        for (const file of files) {
+            const fileStat = await fsCommon.stat(file)
```

Comment:
collectFiles already calls `fsCommon.stat` on each file, but if it calls this then that doubles the cost. Is it possible to couple this with the logic in collectFiles ?